src/jllvm/vm/Interpreter.cpp

```diff
@@ -978,18 +992,13 @@ std::uint64_t jllvm::Interpreter::executeMethod(const Method& method, std::uint1
                 }
                 return ReturnValue(value);
             },
-            [&](InstanceOf instanceOf)
+            [&](OneOf<JSR, JSRw> jsr)
             {
-                auto* object = context.pop<ObjectInterface*>();
-                if (!object)
-                {
-                    context.push<std::int32_t>(0);
-                    return NextPC{};
-                }
-
-                ClassObject* classObject = getClassObject(classFile, instanceOf.index);
-                context.push<std::int32_t>(object->instanceOf(classObject));
-                return NextPC{};
+                std::uint16_t retAddress =
+                    jsr.offset + sizeof(OpCodes)
+                    + (holds_alternative<JSRw>(*curr) ? sizeof(std::int32_t) : sizeof(std::int16_t));
+                context.push(reinterpret_cast<ObjectInterface*>(static_cast<std::uintptr_t>(retAddress)));
```

Comment:
Pushing this as `ObjectInterface*` is problematic as the GC will think its an object reference.
I am guessing you did so due to the JVM spec basically requiring that `astore` is used to store it in a local?

I think the right way to do this here would be to use `pushRaw`, push it as the zero extended integer but make sure that it the GC mask bit is NOT set. 
Then change the implementation of `astore` to not set the GC mask bit for the local if we the value being popped from the operand stack doesn't have it set either (this can done with `popRaw`). You'll probably need to add an overload to `setLocal` that sets the GC mask based on parameter (or maybe just accept a `RawValue`)